### PR TITLE
Remove deprecated deriveAddress from Client

### PR DESCRIPTION
**Pull request by JST5000**
## High Level Overview of Change

* Remove deriveAddress from Client
* Remove alias in client for deriveXAddress (So the workflow is just importing derive from the utils folder now)
* Move the deriveXAddress test from the Client to Utils folder

### Context of Change

Follow up to #1534 for deriveAddress and deriveXAddress.

### Type of Change

- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Before / After

Instead of using Client to access these functions, now you'll have to import them directly from the derive.ts util.

## Test Plan

CI Passes

### CODE REVIEW

**Review by natenichols — APPROVED** [review]
LGTM 👍

**Review by khancode — APPROVED** [review]

File: src/client/index.ts [hunk 1]
```diff
@@ -109,7 +109,6 @@ import {
 import { BaseRequest, BaseResponse } from '../models/methods/baseMethod'
 import combine from '../transaction/combine'
 import { sign } from '../transaction/sign'
-import { deriveAddress, deriveXAddress } from '../utils/derive'
```
Comment (mvadari) [inline src/client/index.ts:112]: If `deriveAddress` is being removed, can we remove the code from `'../utils/derive'`?

**Review by mvadari — CHANGES_REQUESTED** [review]
see comment

File: src/client/index.ts [hunk 2]
```diff
@@ -203,6 +202,7 @@ class Client extends EventEmitter {
    * @param server - URL of the server to connect to.
    * @param options - Options for client settings.
    */
+  // eslint-disable-next-line max-lines-per-function -- Client needs to space for initializing subscriptions
```
Comment (mvadari) [inline src/client/index.ts:205]: don't bother adding these eslint comments - the rule will be edited a bit later and that'll handle this error

**Review by mvadari — APPROVED** [review]

**Pull request merged**